documentation/src/main/asciidoc/topics/marshalling.adoc

```diff
@@ -1,317 +1,173 @@
+{brandname} uses marshalling/unmarshalling to:
 
 * Transform data so that it can be send over to other {brandname} nodes in a cluster.
 * Transform data so that it can be stored in underlying cache stores.
-* Store data in {brandname} in a wire format to provide lazy deserialization capabilities.
-
-== The Role Of JBoss Marshalling
-Since performance is a very important factor in this process, {brandname} uses JBoss Marshalling framework instead of standard Java Serialization in order to marshall/unmarshall Java POJOs. Amongst other things, this framework enables {brandname} to provide highly efficient ways to marshall internal {brandname} Java POJOs that are constantly used. Apart from providing more efficient ways to marshall Java POJOs, including internal Java classes, JBoss Marshalling uses highly performant `java.io.ObjectOutput` and `java.io.ObjectInput` implementations compared to standard `java.io.ObjectOutputStream` and `java.io.ObjectInputStream`.
-
-== Support For Non-Serializable Objects
-From a users perspective, a very common concern is whether {brandname} supports storing non-Serializable objects. In 4.0, an {brandname} cache instance can only store non-Serializable key or value objects if, and only if:
-
-* cache is configured to be a local cache _and..._
-* cache is not configured with lazy serialization _and..._
-* cache is not configured with any write-behind cache store
+* Store data in {brandname} in a binary format to provide lazy deserialization capabilities.
 
-If either of these options is true, key/value pairs in the cache will need to be marshalled and currently they require to either to extend `java.io.Serializable` or `java.io.Externalizable`.
+{brandname} handles marshalling for all internal types. Users only need to be concerned with the marshalling
+of the Java objects that they will store in the cache. The default marshaller for user types is based upon the {brandname}
+library link:#protostream[ProtoStream], however we provide also alternative marshallers such as a link:#java_serialization_marshaller[Serialization based marshaller].
+Furthermore, it's also possible to define a custom link:{javadocroot}/org/infinispan/commons/marshall/Marshaller.html[marshaller implementation]
+for user types if required.
 
-TIP: Since {brandname} 5.0, marshalling non-Serializable key/value objects are supported as long as users can to provide meaningful Externalizer implementations for these non-Seralizable objects.
+== Marshaller Implementations
+[[protostream_cm_config]]
+=== ProtoStream (Default)
+The default marshaller for {brandname} is link:#protostream[ProtoStream], which marshalls data as link:https://developers.google.com/protocol-buffers[Protocol Buffers]. This is a platform independent format
+that utilises schemas to provide a structured representation of your Java objects in a manner that can be easily evolved
+over time whilst maintaining backwards compatibility.
```

Comment:
Possibly "a structured representation of your Java objects that you can easily expand and adapt for changing requirements while maintaining backwards compatibility." 

in any case s/whilst/while - for better or worse US English is the house style

Reply:
I think `evolve` actually makes more sense in this context, because you can't really change/adapt what is already there, rather add to it and then ignore the old fields in the implementation.

Reply:
gotcha. I figured there were some nuances in there I wasn't getting. to me evolve means to change over time in an organic way. I was trying to go for a phrasing to suggest users would be actively modifying.

"objects that can be easily evolved" is a bit of an awkward structure though. it's passive tense so there is an implicit "by <the actor>".

imagine: "objects that can be easily evolved by you over time whilst maintaining backwards compatibility" 

so part of my original comment is also switching it to the active voice and trying to put the focus on the reader. how about this?

"a structured representation of your Java objects that can evolve over time but also maintain backwards compatibility"